### PR TITLE
Qualify link context

### PR DESCRIPTION
**Pull request by frozenfrank**
## TL;DR
Google now recommends using `rel="ugc"` as a more specific edition of `rel="nofollow"`. This PR adjusts the link transformer to follow the recommendation.

## Why is this necessary?
While the `rel="nofollow"` attribute continues to work just fine since it's introduction to the world in 2005 by Google, Google now recommends being even more specific with `rel` attribute. Since 2020, they have recommended using the `rel="ugc"` attribute to _more specifically_ indicate that the link originates from a user. 

Read the official Google docs on this [qualifying outbound links](https://developers.google.com/search/docs/crawling-indexing/qualify-outbound-links), or read more about the context of the decision on [conductor.com](https://www.conductor.com/academy/nofollow/#other-values-ugc-and-sponsored).

## What is changing?
Previously, the `MessageComponent:wrapLinskWithAnchorTag()` private method would generate anchor tags in the following form: `<a ... rel="nofollow">`.

Now, this is adapted to include **both** the "ugc" and the "nofollow" attribute values (`<a ... rel="ugc nofollow">`). While the Google Search docs above specifically say that using the "ugc" value alone is sufficient, I assume that StreamChat is targeting the widest possible accessibility limits, and using both values should only be more acceptable.

I also attempted to update the tests to match this change. However, without documentation on how to run the tests myself, I wasn't able to confirm that they still pass.

### CODE REVIEW

**Comment by szuperaz** [comment]
Thank you for the detailed explanation. From the link you shared: "Currently, the search engine suggests that the “sponsored” value be used for all forms of advertisements or paid placement, and “ugc” for user-generated content (UGC) links, such as comments and forum posts." -> it seems to me that `ugc` should be used if the link points to user-generated content, which is not guaranteed.

To me the `nofollow` seems like a more sensible default. From the link you shared: "As for the nofollow link attribute, Google now suggests you use it for links which are neither sponsored nor user-generated but which still need a signal that you don’t want to be associated with the linked website, or that you don’t want Googlebot to crawl the linked pages."

But I understand the need to set these attributes, so for better flexibility, I've created an issue to provide a custom template for links: https://github.com/GetStream/stream-chat-angular/issues/560 - [it seems GH is having some issues](https://www.githubstatus.com/), but as soon as that is resolved, that feature will be released.
